Adamant/Services/FilesNetworkManager/IPFSApiService.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  IPFSApiService.swift
+//  Adamant
+//
+//  Created by Stanislav Jelezoglo on 09.04.2024.
+//  Copyright © 2024 Adamant. All rights reserved.
+//
+
+import Foundation
+import CommonKit
+
+enum IPFSApiCommands {
+    static let file = (
+        upload: "api/file/upload",
+        download: "api/file/",
+        fieldName: "files"
+    )
+}
+
+final class IPFSApiService: FileApiServiceProtocol {
+    let service: BlockchainHealthCheckWrapper<IPFSApiCore>
+    
+    init(
+        healthCheckWrapper: BlockchainHealthCheckWrapper<IPFSApiCore>
+    ) {
+        service = healthCheckWrapper
+    }
+    
+    func request<Output>(
+        _ request: @Sendable (APICoreProtocol, Node) async -> ApiServiceResult<Output>
+    ) async -> ApiServiceResult<Output> {
+        await service.request { admApiCore, node in
+            await request(admApiCore.apiCore, node)
+        }
+    }
+    
+    func uploadFile(
+        data: Data,
+        uploadProgress: @escaping ((Progress) -> Void)
+    ) async throws -> String {
```

Comment:
Due to unclearness of what type of error it returns I suggest to not use this approach. Also, in the past we changed many method to get rid of this approach.

Let's use this approach: `func request(args...) async -> SomeResult<Output, SomeCustomError>`

It will allow us to see all the possible errors that can be potentially returned